impeller/aiks/paint.h

```diff
@@ -36,6 +36,16 @@ struct Paint {
     kStroke,
   };
 
+  enum class ColorSourceType {
```

Comment:
Feels like with the enum + closure that the current color source abstraction is stretched a little thin. Perhaps we should collect both into a new object on paint? 

I'm not sure where we are on re-using DLtypes but that seems like the best candidate

Reply:
I agree and have some ideas about this -- mind if I resolve it in a followup patch? I think it's time we wrap up all these factories in setColorSource and ditch the two-step conversion.

Reply:
Fine with me!